pkg/admission/doc.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+Package admission provides methods to ensure the webhooks can have
+proper CA and certificate to work correctly.
+
+There are 3 typical ways to use this library:
+
+* Deploying as a controller. Sync method can easily wrapped as a Reconcile method.
+
+* Invoking it as a library in the webhook server before starting to serve the traffic.
+
+* Deploying it as an init container along with the webhook server.
+
+Webhook Configuration
+
+The following is an example MutatingWebhookConfiguration in yaml.
+
+	apiVersion: admissionregistration.k8s.io/v1beta1
+	kind: MutatingWebhookConfiguration
+	metadata:
+	  name: myMutatingWebhookConfiguration
+	  annotations:
+	    secret.certprovisioner.kubernetes.io/webhook-1: namespace-bar/secret-foo
+	    secret.certprovisioner.kubernetes.io/webhook-2: default/secret-baz
+	webhooks:
+	- name: webhook-1
+	  rules:
+	  - apiGroups:
+		- ""
+		apiVersions:
+		- v1
+		operations:
+		- "*"
+		resources:
+		- pods
+	  clientConfig:
+		service:
+		  namespace: service-ns-1
+		  name: service-foo
+		  path: "/mutating-pods"
+		caBundle: [] # CA bundle here
+	- name: webhook-2
+	  rules:
+	  - apiGroups:
+		- apps
+		apiVersions:
+		- v1
+		operations:
+		- "*"
+		resources:
+		- deployments
+	  clientConfig:
+		service:
+		  namespace: service-ns-2
+		  name: service-bar
+		  path: "/mutating-deployment"
+		caBundle: [] # CA bundle here
+
+Build the CertProvisioner
+
+You can choose to build your own Options with the CertInput and the certoutput.Options.
+Or an easier way is to use an empty Options the package will default it with reasonable values.
+The package will write self-signed certificates to secrets.
+
+	// Build a CertProvisioner with empty options
+	cp := &CertProvisioner{Client: getClientOrDie(), Options: Options{}}
```

Comment:
They shouldn't have to specify Options.  I don't think the getClientOrDie() function is correct.  Use the one from the `config` package